src/main/kotlin/com/exactpro/th2/httpserver/api/impl/BasicResponseManager.kt

```diff
@@ -52,38 +53,33 @@ class BasicResponseManager : IResponseManager {
 
     override fun handleRequest(request: RawHttpRequest, answer: (RawHttpResponse<*>) -> Unit) {
         val uuid = UUID.randomUUID().toString()
-        val sequence = generateSequence()
-        dialogs[uuid] = answer
+        dialogs[uuid] = RequestData(request, answer)
         logger.debug("Stored dialog: $uuid")
-        val messageGroup = request.toBatch(context.connectionID, sequence, uuid)
-        context.messageRouter.sendAll(messageGroup, QueueAttribute.SECOND.toString())
+        publishMessage(request, uuid)
         logger.debug("Send on alias: ${context.connectionID.sessionAlias}")
     }
 
     override fun handleResponse(messages: MessageGroup) {
         val response = Th2Response.Builder().setGroup(messages).build()
-        if (response.uuid == null) {
-            throw IllegalArgumentException("UUID is required")
-        }
-        dialogs[response.uuid]?.let {
-            it(response)
-        } ?: run {
-            throw IllegalArgumentException("UUID is not present")
-        }
+        val data = dialogs[response.uuid] ?: throw IllegalArgumentException("UUID is not present in store")
+        data.answer(response)
+        publishMessage(data.request, response)
     }
 
     override fun close() {
 
     }
 
-    /*
-    private val DATE_HEADER_PROVIDER = DateHeaderProvider(Duration.ofSeconds(1))
+    private fun publishMessage(request: RawHttpRequest, uuid: String) {
+        context.messageRouter.sendAll(
+            request.toBatch(context.connectionID, generateSequenceRequest(), uuid),
```

Comment:
We should assign `context.messageRouter` / `context.connectionID` to dedicated variables at `init` to avoid retrieving them from context every time